src/psyclone/tests/gocean1p0_opencl_test.py

```diff
@@ -175,18 +182,33 @@ def test_invoke_opencl_initialisation(kernel_outputdir):
         num_cmd_queues = get_num_cmd_queues()
         cmd_queues => get_cmd_queues()
         kernel_compute_cu_code = get_kernel_by_name("compute_cu_code")
+        call initialise_device_buffer(cu_fld)
+        call initialise_device_buffer(p_fld)
+        call initialise_device_buffer(u_fld)
+        xstart = cu_fld%internal%xstart
+        xstop = cu_fld%internal%xstop
+        ystart = cu_fld%internal%ystart
+        ystop = cu_fld%internal%ystop
+      cu_fld_cl_mem = transfer(cu_fld%device_ptr, cu_fld_cl_mem)
```

Comment:
After a quick exploration I couldn't find the reason, I don't see anything about indentation on f2pygen but I see it creates the expected (fparser1?) tree. I assume the indentation logic happens inside fparser1? Do you have any pointers where to look at?

Reply:
Having dug into this a bit I think it's probably because `PSyIRGen` does a generic parse of the generated Fortran (and thus generates an 'entire' parse tree) while other classes in f2pygen just create a specific fparser1 Statement. This then means that the fparser1 (sub-class of) Statement that is given to the `BaseGen.__init__` has a 'parent' Statement and that means that `fparser.common.base_classes.Statement.get_indent_tab()` adds indentation spacing. Possibly we could 'fix' this by setting the parent of the Statement to None? (As an aside, does PSyIRGen work for multi-statement code fragments?)

Reply:
If I understood you correctly, I nulled the parent of the fparser node obtained by parsing the content provided to PSyIRGen, but it didn't solve the indentation. Which has me a bit confused because the two lines
```
ystop = cu_fld%internal%ystop
cu_fld_cl_mem = TRANSFER(cu_fld%device_ptr, cu_fld_cl_mem)
```
are both provided to the `PSyIRGen.__init__()` with the same parent, and inspecting this method the fortran_writer returns very similar lines for both of them, but then fparser.parser() does already show a different indentation in their "content" section, before they are added to the f2pygen tree at all.
```
(Pdb) fortran_writer(content)
'ystop = cu_fld%internal%ystop\n'
(Pdb) fparser1_parser.block
BeginSource
  blocktype='beginsource'
  name='<_io.StringIO object at 0x7fd505ba2dc0> mode=pyf'
  content:
          Assignment
            variable='ystop'
            sign='='
            expr='cu_fld%internal%ystop'
            item=Line('ystop = cu_fld%internal%ystop',(1, 1),None,None,<reader>)
```
```
(Pdb) fortran_writer(content)
'cu_fld_cl_mem = TRANSFER(cu_fld%device_ptr, cu_fld_cl_mem)\n'
(Pdb) fparser1_parser.block
BeginSource
  blocktype='beginsource'
  name='<_io.StringIO object at 0x7fb3b874fd30> mode=pyf'
  content:
    Assignment
      variable='cu_fld_cl_mem'
      sign='='
      expr='TRANSFER(cu_fld%device_ptr, cu_fld_cl_mem)'
      item=Line('cu_fld_cl_mem = TRANSFER(cu_fld%device_ptr, cu_fld_cl_mem)',(1, 1),None,None,<reader>)
```
Anyway, #1134 should get rid of all these.